### ISSUE TITLE
Error installing with Poetry on Python 3.12 under Homebrew on MacOS

### ISSUE THREAD

**Issue by francispotter** [ludwigschwardt/python-gnureadline#65]
As others have pointed out, proper `readline` support seems to have disappeared from Python under Homebrew, and `gnureadline` provided an excellent alternative, which I used every day, until...

My projects that use Poetry can't build under Python 3.12.

- `pip install gnureadline` works under Python 3.12
- `poetry install` works under Python 3.11

**[Full experiment including error message](https://gitlab.com/steampunk-wizard/sandbox/gnureadline/)**

From [`pyproject.toml`](https://gitlab.com/steampunk-wizard/sandbox/gnureadline/-/blob/main/pyproject.toml):

```
[tool.poetry.dependencies]
python = "^3.12"
gnureadline = "^8.1.2"
```

I admit to zero experience with C library development, but I posted the [whole output](https://gitlab.com/steampunk-wizard/sandbox/gnureadline/-/blob/main/ERROR_MESSAGE.txt) from `poetry install` including the voluminous error messages.

This line looks important:

`Note: This error originates from the build backend, and is likely not a problem with poetry but with gnureadline (8.1.2) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "gnureadline (==8.1.2)"'.`

I ran the `pip wheel` command suggested above and received an equally volumious [2nd error message](https://gitlab.com/steampunk-wizard/sandbox/gnureadline/-/blob/main/ERROR_MESSAGE_2.txt).

Can you suggest a workaround?

Thank you!

**Comment by ludwigschwardt** [comment]
Thanks for the report, Francis! The error messages look a lot like the ones Milos found in #62 on Python 3.11, but I have built successfully on 3.11 before, so it's still a bit of a mystery to me.

I recently became a "Poet" too :-), so I should be able to help out there. Thanks for the sandbox - I'll try that out and hopefully get back to you soon.

The [override-readline-via-site-customization](https://github.com/ludwigschwardt/python-gnureadline/tree/override-readline-via-site-customization) branch might also help, as that is my latest thinking.

**Comment by ludwigschwardt** [comment]
I've finally released gnureadline 8.2.10 to PyPI. Thanks again for your patience 😅
